File: app/src/main/java/com/u1fukui/bbs/helper/DialogHelper.java
package com.u1fukui.bbs.helper;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.annotation.StringRes;
import android.support.v7.app.AlertDialog;

import com.u1fukui.bbs.R;

import javax.inject.Inject;

public class DialogHelper {

    public interface ConfirmDialogListener {

        void onClickPositiveButton();

        void onClickNegativeButton();
    }

    private final Context context;

    @Inject
    public DialogHelper(Context context) {
        this.context = context;
    }

    public void showConfirmDialog(@StringRes int titleResId,
                                  @StringRes int messageResId,
                                  @NonNull final ConfirmDialogListener listener) {

        new AlertDialog.Builder(context, R.style.DialogTheme)
                .setTitle(titleResId)
                .setMessage(messageResId)
                .setPositiveButton(android.R.string.ok, (dialogInterface, i) -> listener.onClickPositiveButton())
                .setNegativeButton(android.R.string.cancel, (dialogInterface, i) -> listener.onClickNegativeButton())
                .show();
    }
}

File: app/src/main/java/com/u1fukui/bbs/helper/DialogHelper.kt
package com.u1fukui.bbs.helper

import android.content.Context
import androidx.annotation.StringRes
import androidx.appcompat.app.AlertDialog

import com.u1fukui.bbs.R

import javax.inject.Inject

class DialogHelper @Inject
constructor(private val context: Context) {

    interface ConfirmDialogListener {

        fun onClickPositiveButton()

        fun onClickNegativeButton()
    }

    fun showConfirmDialog(@StringRes titleResId: Int,
                          @StringRes messageResId: Int,
                          listener: ConfirmDialogListener?) {

        AlertDialog.Builder(context, R.style.DialogTheme)
                .setTitle(titleResId)
                .setMessage(messageResId)
                .setPositiveButton(android.R.string.ok) { _, _ -> listener?.onClickPositiveButton() }
                .setNegativeButton(android.R.string.cancel) { _, _ -> listener?.onClickNegativeButton() }
                .show()
    }
}
